src/history/hs_rec.c

```diff
@@ -868,9 +869,11 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
     WT_UNUSED(key);
 #endif
 
-    /* If we will delete all the updates of the key from the history store, we should not reinsert
-     * any update. */
-    WT_ASSERT(session, ts > WT_TS_NONE || !reinsert);
+    /*
+     * If we will delete all the updates of the key from the history store, we should not reinsert
```

Comment:
change "If we will delete" to "If we delete"

Reply:
Fixed.